var/spack/repos/builtin/packages/py-basemap/package.py

```diff
@@ -37,11 +37,23 @@ class PyBasemap(PythonPackage):
     # Per Github issue #3813, setuptools is required at runtime in order
     # to make mpl_toolkits a namespace package that can span multiple
     # directories (i.e., matplotlib and basemap)
-    depends_on('py-setuptools', type=('build', 'run'))
+    depends_on('py-setuptools', type='run')
     depends_on('py-numpy', type=('build', 'run'))
     depends_on('py-matplotlib', type=('build', 'run'))
     depends_on('pil', type=('build', 'run'))
     depends_on('geos')
 
+    def install(self, spec, prefix):
+        # The default `install` method adds `--single-version-external-managed`
+        # because the package depends on py-setuptools. But the package doesn't
+        # actually build with `py-setuptools`, it just needs it at run-time.
+        # Until I can figure out how to query direct build dependencies only,
+        # override the default `install` method.
```

Comment:
You want to look at `dependencies_dict` -- it gets you the immediate dependencies for a spec and allows filtering by deptype.  There's `dependencies_dict`, `dependents_dict`, which get you dicts of specs keyed by name, and there are `dependencies` and `dependents`, which get you list of specs.

Currently these just fetch immediate dependencies, but I would like to make it easier to get different sets of dependencies based on relationship, e.g. it would be nice if you could get direct build dependencies plus their transitive run dependencies. Some regex-like syntax would work well there... e.g. `spec.find_dependncies('br*')`.  I haven't thought through how the expression language would look for that.